src/common/helpers/useForwardedRef.ts

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+const useForwardedRef = <T>(ref: React.ForwardedRef<T>) => {
```

Comment:
Yeah for sure! 

So `useForwardedRef` basically allows us to use a forwarded ref inside a component that receives a forward ref. By use, I mean act on, like focus the input in our case. Normally when you forward a ref, you just pass it right along and don't do anything but make it available to the parent component. But this hook basically merges two refs and keeps them in sync.

useCombinedRefs is kind of similar except it combines 2 refs and keeps them in sync. In our case I wanted to turn a ref we were already using into a forwarded ref available to any parents, so we need to merge the two to keep them in sync.

Reply:
And needed to do this because we need refs to the basic input for height and position calcs on that DOM element, and you can only get a ref to a functional component if you forward the ref. Same reason we needed to do this in tooltip, because combobox can be disabled with a tooltip.